### PR TITLE
Rename "programs" to "contracts"

### PR DESCRIPTION
**Pull request by richardpringle**
The title says it all, doesn't it?

- **Optionally test subdirectories**
- **Don't leave compiler artifacts next to code**
- **Remove unused ignore**
- **Rename `program` to `contract` in runtime**
- **Rename `program` to `contract` in wasmlanche**
- **Rename `program` to `contract` in examples and tests**
- **Convert 'Program' to 'Contract'**
- **Rename files and directories to use "contracts"**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -106,17 +106,17 @@ _The number of blocks that the `hypersdk` stores on-disk, the `AcceptedBlockWind
 to an arbitrary depth (or set to `MaxInt` to keep all blocks). To limit disk IO used to serve blocks over
 the P2P network, `hypervms` can configure `AcceptedBlockWindowCache` to store recent blocks in memory._
 
-### WASM-Based Programs
+### WASM-Based Contracts
 In the `hypersdk`, [smart contracts](https://ethereum.org/en/developers/docs/smart-contracts/)
-(e.g. programs that run on blockchains) are referred to simply as `programs`. `Programs`
+(e.g. contracts that run on blockchains) are referred to simply as `contracts`. `Contracts`
```
Comment (samliok) [inline README.md:111]:
```suggestion
(e.g. smart contracts that run on blockchains) are referred to simply as `contracts`. `Contracts`
```

File: README.md [hunk 2]
```diff
@@ -106,17 +106,17 @@ _The number of blocks that the `hypersdk` stores on-disk, the `AcceptedBlockWind
 to an arbitrary depth (or set to `MaxInt` to keep all blocks). To limit disk IO used to serve blocks over
 the P2P network, `hypervms` can configure `AcceptedBlockWindowCache` to store recent blocks in memory._
 
-### WASM-Based Programs
+### WASM-Based Contracts
```
Comment (samliok) [inline README.md:109]:
```suggestion
### WASM-Based Smart Contracts
```

i think we should use Smart Contracts in the readme(not for everything but important stuff like title)

**Review by samliok — DISMISSED** [review]
the `VMWithContracts` still has references to the old programs verbiage 🤮
https://github.com/ava-labs/hypersdk/blob/e19fb773c2793d3c76c9adb3403ead601e2f657a/examples/vmwithcontracts/actions/publish.go#L19

everything else looks good

File: .gitignore [hunk 3]
```diff
@@ -66,10 +66,7 @@ target/
 Cargo.lock
 **/*.rs.bk
 
-x/programs/simulator/libsimulator.h
-x/programs/runtime/debug
-x/programs/runtime/wasm32-unknown-unknown
-x/programs/runtime/.rustc_info.json
+x/programs/runtime/target
```
Comment (aaronbuchwald) [inline .gitignore:69]: Should this be updated to `x/contracts` ?

**Comment by aaronbuchwald** [comment]
Could we update the programs tutorial dev container as well?

**Review by aaronbuchwald — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 3 above)
Comment (richardpringle) [reply .gitignore:69]: good catch

**Review by aaronbuchwald — APPROVED** [review]

**Pull request merged**